var/spack/repos/builtin/packages/gaudi/package.py

```diff
@@ -35,11 +35,39 @@ class Gaudi(CMakePackage):
 
     maintainers = ["drbenmorgan", "vvolkl"]
 
-    variant("optional", default=False, description="Build most optional components and tests")
+    variant("aida", default=False, description="Build AIDA interfaces support")
+    variant("cppunit", default=False, description="Build with CppUnit unit testing")
     variant("docs", default=False, description="Build documentation with Doxygen")
+    variant("examples", default=False, description="Build examples")
+    variant("gperftools", default=False, description="Build with Google PerfTools support")
+    variant("heppdt", default=False, description="Build with HEP Particle Data Table support")
+    variant("jemalloc", default=False, description="Build with jemalloc allocator support")
+    variant("unwind", default=False, description="Build with unwind call-chains")
     variant("vtune", default=False, description="Build with Intel VTune profiler support")
-
-    # only build subdirectory GaudiExamples when +optional
+    variant("xercesc", default=False, description="Build with Xerces-C XML support")
+
+    # meta-variants
+    optional_variants = [
+        "aida",
+        "cppunit",
+        "examples",
+        "gperftools",
+        "heppdt",
+        "jemalloc",
+        "unwind",
+        "xercesc",
+    ]
+    variant(
+        "optional",
+        default=False,
+        description="Build optional components; equivalent to{}".format(
+            " +".join([""] + optional_variants)
+        ),
+    )
+    for v in optional_variants:
+        conflicts("-" + v, when="+optional")
```

Comment:
Hmm, variant in other packages change frequently, so I wouldn't write these 20 lines but just change the variants.

Reply:
@vvolkl Dropping `optional` ok with you?

Reply:
ok with me!

Reply:
Then we leave this as something local that key4hep or eic spack repos can add in an adhoc way as an overlay.

```suggestion
```